jax/_src/lax/lax.py

```diff
@@ -1543,13 +1549,23 @@ def _nary_lower_mhlo(op: Callable, ctx,
       provided?
   """
   del params
-  aval_out, = ctx.avals_out
-  broadcasted_args = broadcast_mhlo(aval_out, ctx.avals_in, args)
+  avals_in, (aval_out,) = ctx.avals_in, ctx.avals_out
```

Comment:
As per the above comment, we needed to update this function to handle lowering applications like `add x:f32[] y:f32[n]`, by making `broadcast_mhlo` call `DynamicBroadcastInDimOp`. But that means `broadcast_mhlo` needs to know the relevant axis sizes, i.e. the `ir.Value` corresponding to the `n` in the aval. Hence this code substitutes `ir.Value`s into the avals passed as arguments to `broadcast_mhlo`. (An alternative would be to make the `broadcast_mhlo` implementation generate `GetAxisSize`s to get the axis sizes it needs from the arguments themselves, but that is more complex code and seems wasteful.)